ext/mysqli/mysqli_api.c

```diff
@@ -324,19 +326,19 @@ PHP_FUNCTION(mysqli_stmt_bind_param)
 	MYSQLI_FETCH_RESOURCE_STMT(stmt, mysql_stmt, MYSQLI_STATUS_VALID);
 
 	if (!types_len) {
-		php_error_docref(NULL, E_WARNING, "Invalid type or no types specified");
-		RETURN_FALSE;
+		zend_argument_value_error(ERROR_ARG_POS(2), "cannot be empty");
+		RETURN_THROWS();
 	}
 
 	if (types_len != (size_t) argc) {
 		/* number of bind variables doesn't match number of elements in type definition string */
-		php_error_docref(NULL, E_WARNING, "Number of elements in type definition string doesn't match number of bind variables");
-		RETURN_FALSE;
+		zend_argument_count_error("Number of elements in type definition string doesn't match number of bind variables");
```

Comment:
I think a few `the` is missing from this message. :) I would say `The number of elements in the type definition string doesn't match the number of bind variables`, but maybe it's too much now (?). Otherwise, you could say `argument #2 (...)` instead of `type definition string`.

Reply:
Besides, you could use the `... must match the number of bind variables` format, which is the most common wording we use (rather than `... doesn't match ...`).